### PR TITLE
dev/core#2246 Fix failure to filter exports

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fixes a failure to apply limits to member exports per https://lab.civicrm.org/dev/core/-/issues/2246

Before
----------------------------------------
All memberships exported when selecting specific ones

After
----------------------------------------
Only the specific ones are exported

Technical Details
----------------------------------------
This fixes regression caused by us more consistently setting componentTable. Unfortunately it turned out componentTable is a bit of a magic parameter in this code that triggers the first part of an if clause - leading us to miss the else bit,

The short version here is that the presence of a componentClause should take precedences
over the inner join intended as a blunt, and maybe unnecessary, measure to
limit the export to the entity in question

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins retest this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by seamuslee001** [comment]
Jenkins retest this please

**Comment by eileenmcnaughton** [comment]
jenkins you are a grumpy bum

**Comment by seamuslee001** [comment]
Jenkins retest this please

**Comment by seamuslee001** [comment]
I replicated this locally and confirmed that this fixes the problem MOP
